src/pieces/commands/autocommit/git_api.py

```diff
@@ -0,0 +1,50 @@
+import urllib.request
+import json
+from typing import List, Dict,Optional
+from urllib.parse import urlencode
+
+
+
+
+def get_repo_issues(repo_owner:str, repo_name:str) -> List[Optional[Dict[str, str]]]:
+    """
+    This function searches for issues in a public GitHub repository using the search API.
+    Args:
+        repo_owner (str): The owner of the repository.
+        repo_name (str): The name of the repository.
+    Returns:
+        list: A list of dictionaries of opened issues containing basic information about issues (number,title, body).
+        Returns None if no issues are found.
+    """
+    params = {'q': f'is:issue is:open repo:{repo_owner}/{repo_name}', 'per_page': '30'}
```

Comment:
might want to add a TODO to make this dynamic in the future ie add a flag to enable the use to list the number of issues maybe?